servicetalk-test-resources/src/main/resources/junit-platform.properties

```diff
@@ -0,0 +1,5 @@
+junit.jupiter.execution.parallel.enabled = true
+junit.jupiter.execution.parallel.mode.default = same_thread
+junit.jupiter.execution.parallel.mode.classes.default = concurrent
+junit.jupiter.execution.parallel.config.strategy = dynamic
+junit.jupiter.execution.parallel.config.dynamic.factor = 2
```

Comment:
Interesting, I thought that gradle runs tests concurrently because we use `--parallel` option: 
https://github.com/apple/servicetalk/blob/a51d2f275592f7fa2eb1a822ac4964dd24bdba0d/.github/workflows/ci-prb.yml#L45

Two questions I have:
1. Does the parallel execution at jUnit 5 level give us additional benefits to gradle's `--parallel` flag?
2. How does it work for modules that still use jUnit4 but via jUnit5 vintage engine we apply it in the plugin?
https://github.com/apple/servicetalk/blob/ad7e8e1e6abe1bbc024b03c93c376962a87e4be9/servicetalk-gradle-plugin-internal/src/main/groovy/io/servicetalk/gradle/plugin/internal/ServiceTalkLibraryPlugin.groovy#L197-L200

Reply:
1. Gradle `--parallel` is only parallelism at the Gradle task level, it doesn't affect parallelism within a task (at least as far as I can tell). This would mean that javadoc can be built at the same time as spotbugs runs, etc. or any other tasks that don't have an dependency. Trying a build locally using `--parallel`, which I thought was the default based upon seeing multiple Gradle daemons running, the results are **much faster** builds, 5.5 minutes vs about 11-12 minutes.  So I am curious why we use `--parallel` in some of the GitHub action yaml config files, but do not have `org.gradle.parallel=true` in our `gradle.properties` file. If it works well enough for CI it should work in all circumstances.
We should see further build performance improvements by also using test execution parallelism.
2. I do not know what impact it has for JUnit 4. Investigation will be needed.

Reply:
Thanks for #1591!

Read the docs: https://junit.org/junit5/docs/snapshot/user-guide/index.html#writing-tests-parallel-execution
I like that they allow this type of configuration:
```
junit.jupiter.execution.parallel.mode.default = same_thread
junit.jupiter.execution.parallel.mode.classes.default = concurrent
```
This mode makes perfect sense for our tests.

Couple comments:
1. Please, add a link to the jUnit5 docs at the beginning of the file.
2. Does `dynamic.factor=2` provide more speed up? It may make sense for CI to have the factor == 2, but for local development it may consume too much CPU and won't allow developers to do other work when tests are running. Let's start with 1?

Reply:
> Read the docs: https://junit.org/junit5/docs/snapshot/user-guide/index.html#writing-tests-parallel-execution
> I like that they allow this type of configuration:
> 
> ```
> junit.jupiter.execution.parallel.mode.default = same_thread
> junit.jupiter.execution.parallel.mode.classes.default = concurrent
> ```
> 
> This mode makes perfect sense for our tests.
> 
> Couple comments:
> 
> 1. Please, add a link to the jUnit5 docs at the beginning of the file.

Can do.

> 2. Does `dynamic.factor=2` provide more speed up? It may make sense for CI to have the factor == 2, but for local development it may consume too much CPU and won't allow developers to do other work when tests are running. Let's start with 1?

That was from previous experience with JUnit4/Maven parallelization and may have been specific to that project. The tests there had some blocking portions so running at a little higher concurrency kept the CPUs busier. The ServiceTalks tests may not benefit much and you may be right that it would completely saturate a developer system. I will revert to 1 and then do comparisons once the tests are actually running in parallel.